GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -301,59 +303,53 @@ public void executeImport(File file, String serverBase, Map<String, String> crit
     }
 
 
+
     /**
-     * creates a map of where the metadata columns are
+     * returns a map of where the metadata columns are on this sheet
      * @param firstQuestionColumnIndex
      * @return
      */
-    private static Map<String, Integer> calculateMetadataColumnIndex(int firstQuestionColumnIndex, boolean singleOrRepSheet) {
-        Map<String, Integer> metadataColumnIndex = new HashMap<>();
-
-        int currentColumnIndex = -1;
-
-        metadataColumnIndex.put(DATAPOINT_IDENTIFIER_COLUMN_KEY, ++currentColumnIndex);
-
-        if (hasApprovalColumn(firstQuestionColumnIndex, singleOrRepSheet)) {
-            metadataColumnIndex.put(DATAPOINT_APPROVAL_COLUMN_KEY, ++currentColumnIndex);
-        }
-
-        if (hasRepeatIterationColumn(firstQuestionColumnIndex, singleOrRepSheet)) {
-            metadataColumnIndex.put(REPEAT_COLUMN_KEY, ++currentColumnIndex);
-        }
-
-        metadataColumnIndex.put(DATAPOINT_NAME_COLUMN_KEY, ++currentColumnIndex);
-
-        if (hasDeviceIdentifierColumn(firstQuestionColumnIndex, singleOrRepSheet)) {
-            metadataColumnIndex.put(DEVICE_IDENTIFIER_COLUMN_KEY, ++currentColumnIndex);
+    private Map<String, Integer> getMetadataColumnIndex(Sheet sheet, int firstQuestionColumnIndex, int headerRow, boolean singleOrRepSheet) {
+        Map<String, Integer> index = new HashMap<>();
+
+        Row row = sheet.getRow(headerRow);
+        for (int i = 0; i < firstQuestionColumnIndex; i++) {
+            String header = row.getCell(i).getStringCellValue();
+            if (header.equalsIgnoreCase(IDENTIFIER_LABEL)) {
```

Comment:
The tricky part of using the label to determine which column we are dealing with is the possibility that the user inadvertently changes the text or its in another language.  This change would have to go hand in hand with making our documentation clearly state what the column headers should be.

Reply:
Yes. There is also the possibility of locking or colouring the headers in the data_cleaning output to indicate they are important.

Reply:
The issue now lists what needs to change in the documentation. Jana said she would handle that.